signer/kms/signer.go

```diff
@@ -45,15 +49,43 @@ func (s *Signer) Sign(checkName, body string) (string, string, error) {
 		return "", "", signer.ErrNoKeyForCheck
 	}
 
-	if key.Algo != AlgoSHA512 {
-		return "", "", fmt.Errorf("unable to hash algorithm %q, must be SHA512", key.Algo)
+	var digest hash.Hash
+	switch key.Algo {
+	case AlgoSHA256:
+		digest = sha256.New()
+	case AlgoSHA384:
+		digest = sha512.New384()
+	case AlgoSHA512:
+		digest = sha512.New()
+	default:
+		return "", "", fmt.Errorf("Unsupported digest algorithm %v", key.Algo)
+	}
+
+	if _, err := digest.Write([]byte(body)); err != nil {
+		return "", "", err
+	}
+
+	var d kms_pb.Digest
+	switch key.Algo {
```

Comment:
why not group this switch statement with the 1 above so there's only 1 switch statement?

Reply:
My guess is performance! If the algorithm is unsupported, don't bother writing anything. Might I suggest moving this switch statement into the `NewSigner` method, that way we don't create the signer if it'll be misconfigured?

Reply:
Done.